templates/sign_in.html

```diff
@@ -4,58 +4,81 @@
+                </p>
+            </div>
+        </nav>
+    </header>
+    <main class="signup">
 
+        <div class="welcome">
+            <h3 class="welcome-message">
+                {% for message in get_flashed_messages() %}
+                
+                    {% if "Account created" in message or "logged out" in message %}
+                        <div class="alert alert-warning alert-dismissible fade show" role="alert">
+                            {{ message }}
+                            <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
+                        </div>
+                        
+                    {% endif %}
+                {% endfor %}
+            </h3>
+        </div>
+        <div class="signup-container">
+            <h2>Sign In</h2>
+            <form class="signing-in" action="/signing-in" method="post">
+                <!-- <label for="username" class="form-label">Username</label> -->
+                {% for message in get_flashed_messages() %}
+                
+                    {% if "Invalid" in message %}
+                        <div class="alert alert-warning alert-dismissible fade show alert-message" role="alert">
+                            {{ message }}
+                            <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
+                        </div>
+                        
+                    {% endif %}
+                {% endfor %}
+                <div class="username-cont">
+                    <i class="bi bi-person"></i>
+                    <input type="text" class="form-control" id="username1" name="username" placeholder="Username" required>
+                </div>
+    
+                <!-- <label for="password" class="form-label">Password</label> -->
+                {% for message in get_flashed_messages() %}
+                
+                    {% if "Invalid" in message %}
+                        <div class="alert alert-warning alert-dismissible fade show alert-message" role="alert">
+                            {{ message }}
+                            <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
+                        </div>
+                        
+                    {% endif %}
+                {% endfor %}
+                <div class="password-cont">
+                    <i class="bi bi-shield-lock"></i>
+                    <input type="password" class="form-control" id="password" name="password" placeholder="Password"required>
+                </div>
+    
+                <button class="sign-up" type="submit">Sign in</button>
+            </form>
+            <form action="/sign-up" method="get">
+                <p>Don't have an account? Sign up here</p>
+                <button class="sign-up" type="submit">Sign Up</button>
+            </form>
+        </div>
+    
+    </main>
 </body>
+
+
 </html>
```

Comment:
I see you took a lot of time doing this page, My apologies for spaghetti code! Good Job nonetheless!!